storage/postgres/visibility.go

```diff
@@ -29,62 +37,128 @@ type visibilityStorage struct {
 func (vs *visibilityStorage) Create(ctx context.Context, visibility *types.Visibility) (string, error) {
 	v := &Visibility{}
 	v.FromDTO(visibility)
-	return create(ctx, vs.db, visibilityTable, v)
-}
-
-func (vs *visibilityStorage) Get(ctx context.Context, id string) (*types.Visibility, error) {
-	visibility := &Visibility{}
-	if err := get(ctx, vs.db, id, visibilityTable, visibility); err != nil {
-		return nil, err
+	id, err := create(ctx, vs.db, visibilityTable, v)
+	if err != nil {
+		return "", err
 	}
-	return visibility.ToDTO(), nil
+	return id, vs.createLabels(ctx, id, visibility.Labels)
 }
 
-func (vs *visibilityStorage) List(ctx context.Context) ([]*types.Visibility, error) {
-	var visibilities []Visibility
-	err := list(ctx, vs.db, visibilityTable, map[string][]string{}, &visibilities)
-	if err != nil || len(visibilities) == 0 {
-		return []*types.Visibility{}, err
+func (vs *visibilityStorage) createLabels(ctx context.Context, visibilityID string, labels []*types.VisibilityLabel) error {
+	for _, label := range labels {
+		label.ServiceVisibilityID = visibilityID
```

Comment:
when creating labels , createdat/updatedat , ServiceVisibilityID and the label id are all set in different places across the code and this is kind of confusing